### DISCUSSION TITLE
Error handling

### DISCUSSION THREAD

**Discussion by westguard** [UseDataKit/SDK#6] [Ideas]
If an error happens with the `get_data_ids` or `get_row_count` methods, perhaps there could be a generic error exception class (similar to `DataSourceNotFoundException` or `DataNotFoundException`) that can be thrown? I'm thinking if a data source relies on a third party classes, if they throw an exception we should be able to catch those. Perhaps a different solution might be for the API to be able to catch regular exceptions too, for example those thrown by: `throw new \Exception( 'Error message' );`

If we should be using `DataNotFoundException` for generic errors, let me know.

Also, the API seems to be returning a JSON packet if a `DataSourceNotFoundException` or `DataNotFoundException` exception is thrown, but there is currently nothing indicating that an error has occurred on the client-side. Perhaps an error notification should be shown?

**Comment by doekenorg** [comment]
@westguard for generic errors there is a [DataViewException](https://github.com/GravityKit/DataKit/blob/main/src/DataViewException.php) available for now. But indeed error handling should still be added in multiple places. 👍
